src/Plugin.php

```diff
@@ -22,6 +24,22 @@
 final class Plugin implements PluginInterface, EventSubscriberInterface
 {
 
+	/** @var int */
+	private $dirCount;
+
+	public function __construct()
+	{
+		$vendorPos = strpos(__DIR__, 'vendor');
+		if ($vendorPos === false) {
+			return;
+		}
+
+		// walk up to vendor
+		$basePathTail = substr(__DIR__, $vendorPos);
+		// how many parts?
+		$this->dirCount = substr_count($basePathTail, DIRECTORY_SEPARATOR) + 1;
```

Comment:
Actually, we can know that the vendor folder is always `dirname(__DIR__, 3)` here, as this file is in `<vendor-dir>/phpstan/extension-installer/src` whatever the vendor-dir is named.